types/instance.d.ts

```diff
@@ -185,16 +185,16 @@ export interface FastifyInstance<
     RouteGeneric extends RouteGenericInterface = RouteGenericInterface,
     ContextConfig = ContextConfigDefault,
     SchemaCompiler extends FastifySchema = FastifySchema,
-  >(opts: RouteOptions<RawServer, RawRequest, RawReply, RouteGeneric, ContextConfig, SchemaCompiler, TypeProvider>): FastifyInstance<RawServer, RawRequest, RawReply, Logger, TypeProvider>;
-
-  get: RouteShorthandMethod<RawServer, RawRequest, RawReply, TypeProvider>;
-  head: RouteShorthandMethod<RawServer, RawRequest, RawReply, TypeProvider>;
-  post: RouteShorthandMethod<RawServer, RawRequest, RawReply, TypeProvider>;
-  put: RouteShorthandMethod<RawServer, RawRequest, RawReply, TypeProvider>;
-  delete: RouteShorthandMethod<RawServer, RawRequest, RawReply, TypeProvider>;
-  options: RouteShorthandMethod<RawServer, RawRequest, RawReply, TypeProvider>;
-  patch: RouteShorthandMethod<RawServer, RawRequest, RawReply, TypeProvider>;
-  all: RouteShorthandMethod<RawServer, RawRequest, RawReply, TypeProvider>;
+  >(opts: RouteOptions<RawServer, RawRequest, RawReply, RouteGeneric, ContextConfig, SchemaCompiler, TypeProvider, Logger>): FastifyInstance<RawServer, RawRequest, RawReply, Logger, TypeProvider>;
```

Comment:
```suggestion
  >(opts: RouteOptions<RawServer, RawRequest, RawReply, RouteGeneric, ContextConfig, SchemaCompiler, TypeProvider, Logger extends FastifyBaseLogger = FastifyBaseLogger>): FastifyInstance<RawServer, RawRequest, RawReply, Logger, TypeProvider>;
```

Reply:
Shouldnt be this extending from FastifyBaseLogger?

Reply:
> please provide typings tests.

Sure I can look into this! The idea is to extend `tests/types/*.ts` to account for this change, correct?

> Fastify would break, if you have an incomplete logger. Nobody prevents you from extending base logger and pass it to the fastify instance.

Sorry, I'm not sure I follow 😅 Would you mind elaborating?
I can see that Fastify would break if the logger is incomplete, and I am aware of extending the base logger and passing it to the instance - but I do not necessarily see how it relates to this change.
The change here is purely cosmetic - It allows you to specify the used `Logger` type only at the instance level, so that you do not need to specify it with every `.route` you use.